### PR TITLE
Backfill relationship from entity table

### PR DESCRIPTION
**Pull request by zhixuanjia**
Add an endpoint in BaseEntityResource to backfill relationship from entity table.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java [hunk 1]
```diff
@@ -356,6 +357,42 @@ public Task<BackfillResult> backfillEntityTables(@ActionParam(PARAM_URNS) @Nonnu
...
+              Urn source = (Urn) relationship.getClass().getMethod("getSource").invoke(relationship);
+              Urn dest = (Urn) relationship.getClass().getMethod("getDestination").invoke(relationship);
+              BackfillResultRelationship backfillResultRelationship = new BackfillResultRelationship()
+                  .setSource(source)
+                  .setDestination(dest)
+                  .setRemovalOption(relationshipUpdates.getRemovalOption().name())
+                  .setRelationship(relationship.getClass().getSimpleName());
+
+              backfillResult.getRelationships().add(backfillResultRelationship);
+            } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
```
Comment (jsdonn) [inline restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:385]: very minor, but fyi you can throw ReflectiveOperationException which is a superclass of all of these exceptions: https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/lang/ReflectiveOperationException.html

File: dao-api/src/main/java/com/linkedin/metadata/dao/BaseLocalDAO.java [hunk 2]
```diff
@@ -807,7 +808,8 @@ public abstract <ASPECT extends RecordTemplate> void updateLocalIndex(@Nonnull U
    * @param urn the URN for the entity the aspect (which the local relationship is derived from) is attached to
    * @param aspectClass class of the aspect to backfill
    */
-  public abstract <ASPECT extends RecordTemplate> void backfillLocalRelationshipsFromEntityTables(@Nonnull URN urn, @Nonnull Class<ASPECT> aspectClass);
+  public abstract <ASPECT extends RecordTemplate> List<LocalRelationshipUpdates> backfillLocalRelationshipsFromEntityTables(
```
Comment (jsdonn) [inline dao-api/src/main/java/com/linkedin/metadata/dao/BaseLocalDAO.java:811]: nit: update javadocs

**Review by jsdonn — COMMENTED** [review]
minor comments

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java (hunk 1 above)
Comment (zhixuanjia) [reply restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:385]: sounds good.

File: dao-api/src/main/java/com/linkedin/metadata/dao/BaseLocalDAO.java (hunk 2 above)
Comment (zhixuanjia) [reply dao-api/src/main/java/com/linkedin/metadata/dao/BaseLocalDAO.java:812]: updated

**Review by jsdonn — APPROVED** [review]
LGTM

**Pull request merged**